runtime/test_artifacts/keygenerator/keygen.go

```diff
@@ -0,0 +1,66 @@
+package keygenerator
+
+import (
+	"bytes"
+	"encoding/binary"
+	"math/rand"
+	"sync"
+
+	"github.com/pokt-network/pocket/shared/crypto"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+var keygenInitOnce sync.Once
```

Comment:
Optional NIT: we can use factoring here.

Aside: Big fan of this singleton pattern.